From f5c3c0b0e29854a3be0d0343bc4709f3f69ec3fc Mon Sep 17 00:00:00 2001
From: Grygorii Strashko <grygorii.strashko@ti.com>
Date: Thu, 9 Sep 2021 18:02:52 +0300
Subject: [PATCH 2/2] net: hsr: hsr_forward: fix tx_flags copy

Zero-copy flags in tx_flags processed by SKB core API, so do not overwrite
tx_flags in skb copy in case of hsr, copy only time-stamping related bits.

Fixes: 772bcadb720b ("net: hsr: save tx_flags and tag info of tagged skb")
Signed-off-by: Grygorii Strashko <grygorii.strashko@ti.com>
---
 net/hsr/hsr_forward.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/net/hsr/hsr_forward.c b/net/hsr/hsr_forward.c
index 8e8f0ee3fbea..51924125ef0e 100644
--- a/net/hsr/hsr_forward.c
+++ b/net/hsr/hsr_forward.c
@@ -363,7 +363,7 @@ static struct sk_buff *create_tagged_skb(struct sk_buff *skb_o,
 	if (REDINFO_T(skb) == DIRECTED_TX)
 		return skb;
 
-	skb_shinfo(skb)->tx_flags = skb_shinfo(skb_o)->tx_flags;
+	skb_shinfo(skb)->tx_flags |= skb_shinfo(skb_o)->tx_flags & SKBTX_ANY_TSTAMP;
 	skb->sk = skb_o->sk;
 
 	/* TODO: should check socket option instead? */
-- 
2.17.1

